lib/rules/forbid-component-props.js

```diff
@@ -66,12 +71,20 @@ module.exports = {
                     minItems: 1,
                     items: { type: 'string' },
                   },
+                  disallowedForPatterns: {
+                    type: 'array',
+                    uniqueItems: true,
+                    minItems: 1,
+                    items: { type: 'string' },
+                  },
                   message: { type: 'string' },
                 },
-                required: ['disallowedFor'],
+                anyOf: [
```

Comment:
should this be:
```suggestion
                oneOf: [
```
?

Reply:
Not sure, because with `oneOf` you can't use `disallowedFor` and `disallowedForPatterns` simultaneously, while it is still a case. Here is one of the test cases for example: https://github.com/jsx-eslint/eslint-plugin-react/pull/3805/files#diff-0095adf2f1f3e139bb1a1666d7611671ad90ae66f67de0f5cdc7cb8884351968R825-R847

Reply:
@ljharb so, based on the documentation, only one can be successful at a time https://json-schema.org/draft/2020-12/json-schema-core#section-10.2.1.3-2

Reply:
sure. but "only one is required" is the intended semantics, right?

i suppose they're identical semantics, but i read "anyOf" with more optionality than "oneOf", personally ¯\\\_(ツ)_/¯

Reply:
Yeah, semantically about "at least one required" they are identical probably, but "use several of them simultaneously" is not.

Reply:
right - and "any of" implies multiple matches are expected, "one of" implies only one match is expected

Reply:
So, to summarize:
- `anyOf` gives the ability to use several fields like `disallowedFor` and `disallowedForPatterns` at the same time;
- `oneOf` forbids the use of several fields at the same time;

So, here we are, just discussing the difference, or we're considering the replacement with `oneOf`? A bit hard to understand 😄

Reply:
lol i'm saying that i think that the actual behavior will be identical with anyOf or oneOf here, but that `oneOf` seems more appropriate to me.

I won't hold up the PR further on this topic, though.

Reply:
I got it, so, actually, no.
With `oneOf`, if you pass at the same time `disallowedFor` and `disallowedForPatterns ` - it will be a validation error.
With `anyOf` - it won't.